lms/templates/student_account/account_settings.html

```diff
@@ -27,12 +32,17 @@
 
 <%block name="js_extra">
 <%static:require_module module_name="js/student_account/views/account_settings_factory" class_name="AccountSettingsFactory">
-    var fieldsData = ${ json.dumps(fields) };
-    var authData = ${ json.dumps(auth) };
-    var platformName = ${json.dumps(static.get_platform_name())};
+    var fieldsData = ${ fields | n, dump_js_escaped_json };
+    var authData = ${ auth | n, dump_js_escaped_json };
+    var platformName = ${ static.get_platform_name() | n, dump_js_escaped_json };
 
     AccountSettingsFactory(
-        fieldsData, authData, '${user_accounts_api_url}', '${user_preferences_api_url}', ${user.id}, platformName
+        fieldsData, 
+        authData,
+        '${ user_accounts_api_url | n, js_escaped_string }',
+        '${ user_preferences_api_url | n, js_escaped_string }',
```

Comment:
Great questions.  A function like `validate_url()` would only be required if the whole URL is user supplied.  Let me know if that is not clear in the docs.

In terms of hunting down whether the url was properly URL-escaped, I'm thinking that we'll just have to assume it is unless we are working on code where the URL is actually being constructed.  @nedbat - what are your thoughts on this?  @jcdyer I will try to make the docs more clear once we agree.

Also, there is no validate_url() method at this time. :)